pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -211,6 +218,20 @@ func (p *LaunchTemplateProvider) createLaunchTemplate(ctx context.Context, optio
 	return output.LaunchTemplate, nil
 }
 
+func (p *LaunchTemplateProvider) onCacheEvicted(key string, lt interface{}) {
+	p.Lock()
```

Comment:
I don't think we need to lock since the cache eviction is already threadsafe.

Reply:
cache eviction is threadsafe, however onCacheEvicted is not threadsafe. For example, if an LT is evicted from the cache, ensureLaunchTemplate can be executed that will receive a cache miss, and then query LTs and find the LT that was evicted but has not been deleted yet. After it finds that LT, onCacheEvicted can run (if the lock is removed) and delete the LT before it is used which will propagate as a launch failure in Fleet.

Reply:
The lock isn't saving us in this case, since the LT creation has already happened. The only thing saving us from this race is the expiration timeout -- I guess this is fine (necessary?).

Reply:
the lock does save us in the scenario I mentioned. The LT creation I mentioned would occur in the `ensureLaunchTemplate` func which also takes a lock. Locking in both of these funcs ensures that the cache is always consistent with the state of EC2.